doc/dev/custom_devices.rst

```diff
@@ -172,3 +177,126 @@ After doing so, Catalyst should be able to interface with your custom device.
     @qml.qnode(CustomDevice(wires=1))
     def f():
         return measure(0)
+
+
+Below is an example configuration file
+
+.. code-block:: toml
+
+        schema = 1
+
+        [device]
+        name = "dummy.device.qubit"
+        precision = ['float32', 'float64']
```

Comment:
This field is ambiguous. It could be:

1. The number of bits and encoding used to denote a floating point number which may be used by the internal state in the device. E.g., pennylane-lightning can have a quantum state defining each real and imaginary floating point number as either `float32` or `float64`.
2. Support for classical operations where the operands are `float32` or `float64`.
3. Support for parametrized quantum operations where the parameter is `float32` or `float64`.

We are not using this field at the moment, so having a guess of how this field could be used seems a bad decision to me. A better decision in my opinion is to delete this field.

Reply:
I will temporarily add description 1, but if you tell me to delete it, I will be glad to do so.

Reply:
There could be a fourth meaning: the precision returned by measurement processes.

But either way, I think I might agree that this is best deleted. `1.` really seems like something only applicable to simulators, so at best it should live in an "optional", device-specific section.

Reply:
Since we are not using this field at all, I really prefer to remove it entirely. Once we have a clear meaning for it we can add it again.

Reply:
I'll add it to optional if you don't want to remove it.

Reply:
Deleting it is fine by me 👍

Reply:
https://github.com/PennyLaneAI/catalyst/pull/369/commits/eb50cccb875320bff10682bf75e0819d0477205b